integration/createdeletebeforeactive_test.go

```diff
@@ -0,0 +1,83 @@
+// +build integration
+
+package integration_test
+
+import (
+	"fmt"
+
+	awseks "github.com/aws/aws-sdk-go/service/eks"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/weaveworks/eksctl/pkg/ctl/cmdutils"
+	"github.com/weaveworks/eksctl/pkg/testutils/aws"
+	. "github.com/weaveworks/eksctl/pkg/testutils/matchers"
+)
+
+var _ = Describe("(Integration) Create & Delete before Active", func() {
+	const (
+		initNG = "ng-0"
+		testNG = "ng-1"
+	)
+
+	Describe("when creating a cluster with 1 node", func() {
+		var clName string
+		Context("for deleting a creating cluster", func() {
+			It("should run eksctl and not wait for it to finish", func() {
+
+				fmt.Fprintf(GinkgoWriter, "Using kubeconfig: %s\n", kubeconfigPath)
+
+				if clName == "" {
+					clName = cmdutils.ClusterName("", "") + "-delb4active"
+				}
+
+				eksctlStart("create", "cluster",
+					"--verbose", "4",
+					"--name", clName,
+					"--tags", "alpha.eksctl.io/description=eksctl delete before active test",
+					"--nodegroup-name", initNG,
+					"--node-labels", "ng-name="+initNG,
+					"--node-type", "t2.medium",
+					"--nodes", "1",
+					"--region", region,
+					"--version", version,
+				)
+			})
+		})
+
+		Context("when deleting the (creating) cluster", func() {
+
+			It("should not return an error", func() {
+
+				eksctlSuccess("delete", "cluster",
+					"--verbose", "4",
+					"--name", clName,
+					"--region", region,
+					"--wait",
+				)
+			})
+
+			It("and should have deleted the EKS cluster and both CloudFormation stacks", func() {
+
+				awsSession := aws.NewSession(region)
+
+				Expect(awsSession).ToNot(HaveExistingCluster(clName, awseks.ClusterStatusActive, version))
```

Comment:
this doesn't fail if I comment out the deletion above

Reply:
It should now that we properly wait until the cluster shows up as "creating" before subsequently deleting it.

Reply:
I have verified this by commenting out the deletion code and validating that the test fails.